tests/test_plugin.py

```diff
@@ -117,10 +117,38 @@ def test_icon_emoji():
     with mock.patch('requests.post') as mock_post:
         testdir.runpytest('--slack_channel', slack_hook_channel,
                           '--slack_hook', slack_hook_host,
-                          '--slack_success_icon', slack_hook_icon_emoji_success,
-                          '--slack_failed_icon', slack_hook_icon_emoji_failed)
+                          '--slack_success_emoji', slack_hook_icon_emoji_success,
+                          '--slack_failed_emoji', slack_hook_icon_emoji_failed)
 
         called_data = json.loads(mock_post.call_args[1]['data'])
         emoji = called_data['icon_emoji']
 
         assert emoji == expected_emoji
+
+
+@pytest.mark.parametrize('test_input,expected_url', [
+    ('1 == 1', ':sunny:'),
+    ('2 == 1', ':rain_cloud:'),
+])
+def test_pytest_slack_custom_icons(testdir, test_input, expected_url):
+    testdir.makepyfile(
+        """
+        def test_icon_url():
+            assert %s
+        """ % test_input
+    )
+
+    slack_hook_host = 'http://test.com/any_hash'
+    slack_hook_channel = 'test'
+    slack_hook_icon_url_success = ':sunny:'
```

Comment:
Please change from emoji to URL
I understand that this will have no effect on the test.
But it will be more like real supposed usage.